dialect/sql/builder.go

```diff
@@ -1815,6 +1821,38 @@ type Builder struct {
 	total        int           // total number of parameters in query tree.
 }
 
+// PredicateBuilder Limited builder for use in custom predicates
+type PredicateBuilder struct {
```

Comment:
Is there a better way to do this? I tried an interface but could not get it to work